### PR TITLE
Replace custom air.main.basedir property with built-in session.rootDirectory

### PR DESCRIPTION
**Pull request by nineinchnick**
## Description

`session.rootDirectory` was introduced in Maven 3.9

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Comment by nineinchnick** [comment]
It doesn't work with the Modernizer plugin and I'm not sure why.

File: pom.xml [hunk 1]
```diff
@@ -2166,7 +2165,7 @@
                     <artifactId>modernizer-maven-plugin</artifactId>
                     <configuration>
                         <violationsFiles>
-                            <violationsFile>${air.main.basedir}/.mvn/modernizer/violations.xml</violationsFile>
+                            <violationsFile>${session.rootDirectory}/.mvn/modernizer/violations.xml</violationsFile>
```
Comment (cstamas) [inline pom.xml:2168]: It does not work like this, you need an "indirection". This property is evaluated ONLY in Maven CLI params! (CLI or those coming from `maven.config`).

So indirection would be: in maven.config set
`-Dmodernizer.violationsFile=${session.rootDirectory}/.mvn/modernizer/violations.xml`

and in here use
`<violationsFile>modernizer.violationsFile</violationsFile>`

Reason is how Maven4 got many other related changes in this are, but the "backport" to Maven 3.9 was "mild" or "limited". So ONLY MAVEN CLI PARAMS are interpolated. But you can circumvent it with indirection like thiat and it will still work with Maven4 as well.

**Review by cstamas — COMMENTED** [review]
This is a good PR, but modernizer needs a circumvention as described above.

**Comment by cstamas** [comment]
This is a good PR, but modernizer needs a circumvention as described above.

**Comment by wendigo** [comment]
@nineinchnick nice

File: pom.xml (hunk 1 above)
Comment (nineinchnick) [reply pom.xml]: Thanks, let me try that!

**Comment by cstamas** [comment]
`com.github.ekryd.sortpom:sortpom-maven-plugin:3.3.0:verify (default) on project trino-testing-services: The file /home/runner/work/trino/trino/testing/trino-testing-services/pom.xml is not sorted`

So edits did violate some rules, needs reformat?

**Comment by nineinchnick** [comment]
> `com.github.ekryd.sortpom:sortpom-maven-plugin:3.3.0:verify (default) on project trino-testing-services: The file /home/runner/work/trino/trino/testing/trino-testing-services/pom.xml is not sorted`
> 
> So edits did violate some rules, needs reformat?

This tripped on empty `<properties></properties>` tags, I removed them.

**Review by cstamas — APPROVED** [review]

**Comment by cstamas** [comment]
More sortpom issues:
`com.github.ekryd.sortpom:sortpom-maven-plugin:3.3.0:verify (default) on project trino-phoenix5-patched: The file /home/runner/work/trino/trino/lib/trino-phoenix5-patched/pom.xml is not sorted`

**Review by cstamas — APPROVED** [review]

File: client/trino-jdbc/pom.xml [hunk 2]
```diff
@@ -12,7 +12,6 @@
     <artifactId>trino-jdbc</artifactId>
 
     <properties>
-        <air.main.basedir>${project.parent.basedir}</air.main.basedir>
```
Comment (findepi) [inline client/trino-jdbc/pom.xml:15]: Per airbase docs

> For a multi-module project, all other sub-modules must have this explicitly set to the root directory and therefore the following code
> 
> ```xml
> <properties>
>   <air.main.basedir>${project.parent.basedir}</air.main.basedir>
> </properties>
> ```
> must be added to each pom.

https://github.com/airlift/airbase/blob/master/README.md#airmainbasedir

so when using airbase root pom, we are expected to define the `air.main.basedir` property correctly, and this should not be removed

howeever, airbase seems to make no direct use of `air.main.basedir` property, so we may perhaps simply remove it from airbase (remove from airbase pom and readme). this would be a prep step, if we're pedantic.

File: .mvn/maven.config [hunk 3]
```diff
@@ -1,2 +1 @@
 --strict-checksums
-
```
Comment (findepi) [inline .mvn/maven.config:2]: Can you move the first commit to a separate PR to get it merged quicker?

Comment (nineinchnick) [reply .mvn/maven.config:2]: It's related

Comment (nineinchnick) [reply .mvn/maven.config:2]: It doesn't matter how fast it'll get merged

File: client/trino-jdbc/pom.xml (hunk 2 above)
Comment (nineinchnick) [reply client/trino-jdbc/pom.xml:15]: > this would be a prep step, if we're pedantic.

Oh, but we are!

**Review by wendigo — APPROVED** [review]

**Comment by wendigo** [comment]
I like the fact that we are making POMs smaller :)

**Comment by wendigo** [comment]
@nineinchnick do you plan a follow-up in Airbase to remove this property?
